core/esmf-aspect-model-aas-generator/src/main/java/org/eclipse/esmf/aspectmodel/aas/AspectModelAasVisitor.java

```diff
@@ -574,14 +584,15 @@ private <T extends Collection> Environment visitCollectionProperty( final T coll
                         context.getRawPropertyValue()
                               .filter( ArrayNode.class::isInstance )
                               .map( ArrayNode.class::cast )
-                              .map( arrayNode -> ( Property property ) -> {
+                              .map( arrayNode -> ( final Property property ) -> {
                                  final List<SubmodelElement> values = getValues( collection, property, context, arrayNode );
                                  return new DefaultSubmodelElementList.Builder()
                                        .idShort( property.getName() )
                                        .displayName( LangStringMapper.NAME.map( property.getPreferredNames() ) )
                                        .description( LangStringMapper.TEXT.map( property.getDescriptions() ) )
                                        .value( values )
-                                       .typeValueListElement( AasSubmodelElements.SUBMODEL_ELEMENT )
+                                       .typeValueListElement( AasSubmodelElements.SUBMODEL_ELEMENT_COLLECTION )
```

Comment:
Maybe I have a wrong understanding of the property, but if it means "the type of element that are contained within the list", wouldn't `SUBMODEL_ELEMENT_COLLECTION` (SEC) mean that each element is a SEC? Because that's not the case looking at the implementation of `getValues()`

Reply:
the typeValueListElement needs to be determined taking the characteristic into account, i.e. it is only a SubmodelElementCollection if the charcteristic is a SingleEntity. So probably the Case a: xxx b: yyy is missing.